Framework/Kernel/src/NexusDescriptor.cpp

```diff
@@ -271,5 +274,37 @@ void NexusDescriptor::walkFile(::NeXus::File &file, const std::string &rootPath,
   file.closeGroup();
 }
 
+bool NexusDescriptor::findAndOpenParentGroup(::NeXus::File &file,
+                                             const std::string &classType,
+                                             const std::string &className) {
+  const auto entries = file.getEntries();
+  if (!entries.empty()) {
+    auto foundItem =
+        std::find_if(entries.cbegin(), entries.cend(),
+                     [&className, &classType](const auto &entry) {
+                       const auto &[name, type] = entry;
+                       if (className.empty())
+                         return type == classType;
+                       return name == className && type == classType;
+                     });
+
+    if (foundItem != entries.cend()) {
+      return true;
+    } else {
+      for (const auto &[name, type] : entries) {
+        try {
```

Comment:
I'm guessing the try-catch is there because somethings are not groups. Is there a better way to identify aprori `is_group` etc? If not, can we be more specific about the exception type - this could cause future maintenance problems if it's casting too much of a wide net.